stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1415,3 +1415,50 @@ function normalize(v::AbstractVector, p::Real = 2)
         return T[]
     end
 end
+
+# Iterator for the rows() and cols()
+struct MatrixDimIterator{T}
```

Comment:
If you parameterize by the parent array type itself, then you can allow any `AbstractMatrix` to be iterated over — and not just `Matrix`.  This includes things like `ReshapedArray`s and other views — things that we try hard to ensure work as much as possible.  `P` is a common variable for this as this can be thought of as the "parent" array that generates the behavior here.

```julia
struct MatrixDimIterator{P<:AbstractMatrix}
…
    M::P
```